subsys/usb/usb_device.c

```diff
@@ -1538,22 +1540,28 @@ int usb_enable(void)
 	u32_t i;
 	struct usb_dc_ep_cfg_data ep0_cfg;
 
+	/* Prevent from calling usb_enable form different contex.
+	 * This should only be called once.
+	 */
+	LOG_DBG("lock usb_enable_lock mutex");
+	k_mutex_lock(&usb_enable_lock, K_FOREVER);
+
 	if (usb_dev.enabled == true) {
 		return 0;
```

Comment:
No, this will produce deadlocks for anything that subsequently invokes `usb_enable()`.  The body has to be:
```
    ret = 0;
    goto out;
```
where the label `usb_enable_error` is renamed to `out`.

Reply:
Done.